hawc/services/utils/doi.py

```diff
@@ -0,0 +1,41 @@
+import html
+import json
+import urllib.parse
+
+from hawc.apps.lit import constants
+
+
+def get_doi_if_valid(text: str):
+    doi = constants.DOI_EXTRACT.search(text)
+    if doi:
+        doi = doi.group(0)
+        doi = html.unescape(doi)
+        doi = urllib.parse.unquote(doi)
+        while doi.endswith((".", ",", '"')):
```

Comment:
I agree @rabstejnek , but couldn't figure out how. When I added w/ some toy examples the regex wasn't greedy enough; ie., it would stop at the first period, which is often valid.